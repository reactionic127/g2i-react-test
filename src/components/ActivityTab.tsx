import React, { ReactElement, useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { PlusCircleFilled } from "@ant-design/icons";
import AddActivityModal from "./AddActivityModal";
import ActivityList from "./ActivityList";
import { GET_ALL_ACTIVITIES_REQUEST } from "../redux/constants";
import getActivityState from "../redux/selectors";

function ActivityTab(): ReactElement {
  const dispatch = useDispatch();
  const { activities } = useSelector(getActivityState);

  const [modalVisible, setModalVisible] = useState(false);

  useEffect(() => {
    dispatch({ type: GET_ALL_ACTIVITIES_REQUEST });
  }, [dispatch]);

  return (
    <div>
      <button
        data-id="add-button"
        type="button"
        className="create-activity-btn"
        onClick={() => setModalVisible(true)}
      >
        <PlusCircleFilled style={{ fontSize: 22, color: "#4994FF" }} />
        <span className="activity-text">Add new activity</span>
      </button>
      <AddActivityModal
        data-id="add-activity-modal"
        visible={modalVisible}
        setVisible={setModalVisible}
      />
      <ActivityList activities={activities} />
    </div>
  );
}

export default ActivityTab;
