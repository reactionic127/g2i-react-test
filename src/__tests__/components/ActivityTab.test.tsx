import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ActivityTab from "../../components/ActivityTab";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useSelector: () => ({ activities: [] }),
  useDispatch: () => mockDispatch,
}));

jest.spyOn(React, "useEffect").mockImplementation((f) => f());

describe("ActivityTab", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    beforeEach(() => {
      wrapper = shallow(<ActivityTab />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should have activity modal", () => {
      expect(wrapper.find('[data-id="add-activity-modal"]').length).toEqual(1);
    });

    it("simulate button click event", () => {
      wrapper.find('[data-id="add-button"]').simulate("click");
      expect(
        wrapper.find('[data-id="add-activity-modal"]').prop("visible")
      ).toEqual(true);
    });
  });
});
