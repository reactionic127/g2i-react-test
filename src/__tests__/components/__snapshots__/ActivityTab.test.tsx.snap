// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityTab rendering renders the component 1`] = `
<div>
  <button
    className="create-activity-btn"
    data-id="add-button"
    onClick={[Function]}
    type="button"
  >
    <ForwardRef(PlusCircleFilled)
      style={
        Object {
          "color": "#4994FF",
          "fontSize": 22,
        }
      }
    />
    <span
      className="activity-text"
    >
      Add new activity
    </span>
  </button>
  <AddActivityModal
    data-id="add-activity-modal"
    setVisible={[Function]}
    visible={false}
  />
  <ActivityList
    activities={Array []}
  />
</div>
`;
